### PR TITLE
35 library generator add path to workspace tsconfig

### LINKED ISSUE #35: tsconfig path updates
Currently, we include Vite and tsconfig paths for convenient local imports, e.g. `@app/` and `@assets/`. In a large Nx project this becomes unwieldy, since tsconfig paths arrays are not merged with `tsconfig.base.json`. We end up needing to duplicate all paths in `tsconfig.app.json` as well as `vite.config.ts`.

We'll either need to switch to relative imports in the app, or use paths from `tsconfig.base.json`. To alleviate the burden of manually including each path in `vite.config.ts`, we can do something like:

```ts
import tsconfigBase from '../../tsconfig.base.json'

const resolve = (p: string) => path.resolve(__dirname, p)

const tsconfigBaseAliases = (rootOffset: string): Record<string, string> => {
  const paths = tsconfigBase.compilerOptions?.paths || []
  const aliases: Record<string, string> = {}
  for (const [name, path] of Object.entries(paths)) {
    aliases[name] = resolve(`${rootOffset}/${path}`)
  }
  return aliases
}

export default defineConfig({
  resolve: {
    alias: {
      ...tsconfigBaseAliases('../..'),
    },
  },
}
```

### PR DESCRIPTION
**Pull request by abemscac**
Close #35

### CODE REVIEW

File: e2e/vue3-vite-e2e/tests/library.spec.ts [hunk 1]
```diff
@@ -8,10 +9,6 @@ jest.setTimeout(60000);
 
 describe('library e2e', () => {
   it('should create library', async () => {
-    // Create app
-    const app = uniq('vue3-vite');
-    await runNxCommandAsync(`generate nx-vue3-vite:app ${app}`);
```
Comment (abemscac) [inline e2e/vue3-vite-e2e/tests/library.spec.ts:13]: Remove the app-generation in the e2e test for library generator.
I think we don't really need to generate an app before generate a library.

File: packages/vue3-vite/src/generators/library/generator.ts [hunk 2]
```diff
@@ -128,6 +152,7 @@ export default async function (host: Tree, options: LibraryGeneratorSchema) {
     tags: normalizedOptions.parsedTags,
   });
 
+  ensureRootFiles(host, normalizedOptions);
```
Comment (abemscac) [inline packages/vue3-vite/src/generators/library/generator.ts:155]: Call a new function here to add path to `tsconfig.base.json`.

**Comment by abemscac** [comment]
Hi @sampullman 

I'm not sure why unit tests in CI environment failed, but passes on local environment. 🤔 
It seems that `getWorkspaceLayout(host).libsDir` is resolved as empty string in CI environment, which leads to the failure we see.

File: e2e/vue3-vite-e2e/tests/library.spec.ts (hunk 1 above)
Comment (sampullman) [reply e2e/vue3-vite-e2e/tests/library.spec.ts:13]: Sounds good! I think the only reason to test app+lib together is to make sure they interact correctly (build, serve, etc). But most of the E2E tests don't need that.

**Comment by sampullman** [comment]
I tracked down the problem! Nx calculates `libsDir` as:
```ts
{
  // etc...
  libsDir:
      nxJson?.workspaceLayout?.libsDir ??
      inOrderOfPreference(tree, ['libs', 'packages'], '.'),
}

function inOrderOfPreference(
  tree: Tree,
  selectedFolders: string[],
  defaultChoice: string
) {
  for (let i = 0; i < selectedFolders.length; ++i) {
    if (tree.exists(selectedFolders[i])) return selectedFolders[i];
  }
  return defaultChoice;
}
```

So if `libs` doesn't already exist, it will choose the root directory. You would think `createTreeWithEmptyWorkspace` already takes care of it, but it seems like it doesn't! We need to provide an option: `createTreeWithEmptyWorkspace({ layout: 'apps-libs' })`, and then it works.

Aside from that, everything looks good! I made some changes to how the library generator figures out the directories to more closely match how the `@nrwl/node:lib` generator is doing it now. I noticed there are a lot of new helper functions in `@nrwl/devkit`, so in the future we might want to see if anything else can be simplified.

File: packages/vue3-vite/src/generators/library/generator.ts [hunk 3]
```diff
@@ -31,20 +37,28 @@ function normalizeOptions(
...
-    ? names(options.directory).fileName
-    : fileName;
-  const libraryRoot = joinPathFragments(
-    getWorkspaceLayout(host).libsDir,
-    libraryDirectory
-  );
-  const parsedTags = options.tags
-    ? options.tags.split(',').map((s) => s.trim())
-    : [];
+  const { layoutDirectory, projectDirectory: libraryDirectory } =
```
Comment (sampullman) [inline packages/vue3-vite/src/generators/library/generator.ts:40]: The following lines are almost identical to the `@nrwl/node:lib` generator.

File: packages/vue3-vite/src/generators/library/generator.ts [hunk 4]
```diff
@@ -31,20 +37,28 @@ function normalizeOptions(
...
+    : name;
+
+  const libraryName = fullLibraryDirectory.replace(new RegExp('/', 'g'), '-');
+  const fileName = getCaseAwareFileName({
+    fileName: libraryName,
+    pascalCaseFiles: options.pascalCaseFiles,
+  });
+  const libraryRoot = joinPathFragments(libsDir, fullLibraryDirectory);
+
+  const parsedTags = parseTags(options.tags);
```
Comment (sampullman) [inline packages/vue3-vite/src/generators/library/generator.ts:57]: Added a helper function for parsing tags

File: packages/vue3-vite/src/generators/library/schema.d.ts [hunk 5]
```diff
@@ -1,5 +1,6 @@
 export interface LibraryGeneratorSchema {
   name: string;
+  pascalCaseFiles?: boolean;
```
Comment (sampullman) [inline packages/vue3-vite/src/generators/library/schema.d.ts:3]: Added an option for file casing, similar to `@nrwl/node:lib`

**Comment by abemscac** [comment]
Hi @sampullman 

Sorry! I just realized that this PR is actually for #26, not #35! 🤦 
I sincerely apologize for any inconvenience caused.

I think we should reopen #35, and close #26 instead.

**Comment by sampullman** [comment]
Whoops! I think I noticed that when I first saw the PR, but forgot to change the linked issue.

**Pull request merged**